packages/fastify-renderer/src/node/renderers/react/ReactRenderer.tsx

```diff
@@ -407,11 +416,23 @@ export class ReactRenderer implements Renderer {
           const url = new URL('fstr://' + id)
           const lazy = !!url.searchParams.get('lazy')!
           const base = url.searchParams.get('base')!
-          const applicableRoutes = this.routes.filter((route) => route.base == base)
-          applicableRoutes.sort((a, b) => routeSortScore(a.url) - routeSortScore(b.url))
+          const route = url.searchParams.get('route')!
+          const component = url.searchParams.get('component')!
+          // We filter out the routes the imperatively renderable routes, which don't have a url property
+          // There is no point in having them included in the route table
+          const applicableRoutes = this.routes.filter((route) => route.base == base && route.url !== undefined)
+          if (route) {
+            const imperativeRoute = Object.assign(
+              {},
+              this.routes.find((route) => (route.renderable = component)),
```

Comment:
Should this be a `==`? And if so, is there a missing test that would prove this is working?